### PR TITLE
Add an example for StreamProvider #2004

### PR DESCRIPTION
**Pull request by huzaifansari54**
stream provider docs update  with example it is 2nd PR

### CODE REVIEW

**Comment by huzaifansari54** [comment]
this exapmle if you wants any change let me know

File: website/docs/providers/stream_provider/live_stream_chat_consumer.dart [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+import 'live_stream_chat_provider.dart';
+
+/* SNIPPET START */
+// UI part is same as like Future provider
```
Comment (rrousselGit) [inline website/docs/providers/stream_provider/live_stream_chat_consumer.dart:7]:
```suggestion
```

Comment (rrousselGit) [inline website/docs/providers/stream_provider/live_stream_chat_consumer.dart:7]:
```suggestion
```

File: website/docs/providers/stream_provider/live_stream_chat_consumer.dart [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+Widget build(BuildContext context, WidgetRef ref) {
+  final liveChats = ref.watch(liveStreamProvider);
+  return liveChats.when(
```
Comment (rrousselGit) [inline website/docs/providers/stream_provider/live_stream_chat_consumer.dart:10]:
```suggestion
  // Like FutureProvider, it is possible to handle loading/error states using AsyncValue.when
  return liveChats.when(
```

File: website/docs/providers/stream_provider/live_stream_chat_provider.dart [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    yield await Future.delayed(const Duration(seconds: 1));
+    try {
+      // yield chats
+      yield fakeChats[i] as String;
+    } catch (error) {
+      // handling error
+      yield* Stream.error(error);
+    }
+  }
+}
```
Comment (rrousselGit) [inline website/docs/providers/stream_provider/live_stream_chat_provider.dart:26]: No reason to separate these two

```suggestion
final chatProvider = StreamProvider<List<String>>((ref) async* {
  // Connect to an API using sockets, and decode the output
  final socket = await Socket.connect('my-api', 4242);
  ref.onDispose(socket.close);
  
  var allMessages = const <String>[];
  await for (final message in socket.map(utf8.decode)) {
    // A new message has been received. Let's add it to the list of all messages.
    allMessages = [...allMessages, message];
    yield allMessages;
  }
});
```

File: website/docs/providers/stream_provider/live_stream_chat_provider.dart [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import 'dart:async';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
```
Comment (rrousselGit) [inline website/docs/providers/stream_provider/live_stream_chat_provider.dart:2]:
```suggestion
import 'dart:io';

import 'package:flutter_riverpod/flutter_riverpod.dart';
```

File: website/docs/providers/stream_provider/live_stream_chat_provider.dart [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+/* SNIPPET START */
+// Fake chats as data sorce
+// Error last index is represent error which type is not a string
+const fakeChats = ['John Hi', 'Flutter is Cute', 1];
+
+// stream provider
```
Comment (rrousselGit) [inline website/docs/providers/stream_provider/live_stream_chat_provider.dart:9]:
```suggestion
```

File: website/docs/providers/stream_provider/live_stream_chat_consumer.dart [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+import 'live_stream_chat_provider.dart';
+
+/* SNIPPET START */
+// UI part is same as like Future provider
+Widget build(BuildContext context, WidgetRef ref) {
+  final liveChats = ref.watch(liveStreamProvider);
```
Comment (rrousselGit) [inline website/docs/providers/stream_provider/live_stream_chat_consumer.dart:9]:
```suggestion
Widget build(BuildContext context, WidgetRef ref) {
  final liveChats = ref.watch(chatProvider);
```

File: website/docs/providers/stream_provider/live_stream_chat_consumer.dart [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  return liveChats.when(
+    loading: () => const CircularProgressIndicator(),
+    error: (error, stackTrace) => Text(error.toString()),
+    data: (msgs) {
```
Comment (rrousselGit) [inline website/docs/providers/stream_provider/live_stream_chat_consumer.dart:13]:
```suggestion
    data: (messages) {
```

File: website/docs/providers/stream_provider/live_stream_chat_consumer.dart [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+      return Text(msgs);
```
Comment (rrousselGit) [inline website/docs/providers/stream_provider/live_stream_chat_consumer.dart:14]:
```suggestion
      // Display all the messages in a scrollable list view.
      return ListView.builder(
        // Show messages from bottom to top
        reverse: true,
        itemCount: messages.length,
        itemBuilder: (context, index) {
          final message = messages[index];
          return Text(message);
        },
      );
```

File: website/docs/providers/stream_provider.mdx [hunk 9]
```diff
@@ -42,3 +44,14 @@ Using `StreamProvider` over [StreamBuilder] has numerous benefits:
 [stream.periodic]: https://api.dart.dev/stable/2.15.1/dart-async/Stream/Stream.periodic.html
 [family]: ../concepts/modifiers/family
 [streambuilder]: https://api.flutter.dev/flutter/widgets/StreamBuilder-class.html
+
+## Usage example: listening live chats from server
```
Comment (rrousselGit) [inline website/docs/providers/stream_provider.mdx:48]:
```suggestion
## Usage example: live chat using sockets
```

**Review by rrousselGit — APPROVED** [review]

**Comment by rrousselGit** [comment]
Thanks! I took the liberty to change the example a bit, to use websockets and show a list of messages instead of just the latest message

File: website/docs/providers/stream_provider/live_stream_chat_provider.dart [hunk 10]
```diff
@@ -0,0 +1,18 @@
+import 'dart:async';
```
Comment (rrousselGit) [inline website/docs/providers/stream_provider/live_stream_chat_provider.dart:1]:
```suggestion
```

File: website/docs/providers/stream_provider/live_stream_chat_provider.dart [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+/* SNIPPET START */
+final chatProvider = StreamProvider<List<String>>((ref) async* {
+  // Connect to an API using sockets, and decode the output
+  final socket = await Socket.connect('my-api', 4242);
+  ref.onDispose(socket.close);
+  
+  var allMessages = const <String>[];
+  await for (final message in socket.map(utf8.decode)) {
```
Comment (rrousselGit) [inline website/docs/providers/stream_provider/live_stream_chat_provider.dart:13]:
```suggestion
  await for (final message in socket.cast<List<int>>().map(utf8.decode)) {
```

File: website/docs/providers/stream_provider/live_stream_chat_provider.dart [hunk 12]
```diff
@@ -0,0 +1,18 @@
+import 'dart:async';
+import 'dart:io';
```
Comment (rrousselGit) [inline website/docs/providers/stream_provider/live_stream_chat_provider.dart:2]:
```suggestion
import 'dart:convert';
import 'dart:io';
```

**Review by rrousselGit — APPROVED** [review]

**Comment by rrousselGit** [comment]
LGTM, thanks!

**Pull request merged**